tests/test_api_endpoints.py

```diff
@@ -0,0 +1,235 @@
+        expected = {'message': 'Registration succesful, proceed to log in'}
+        # check returned message
+        self.assertEqual(expected, json.loads(response.data))
+
+    def test_can_register_admin(self):
+        '''tests ana dmin can be registered'''
+        response = self.client.post(
+            '/vi/auth/signup', data=json.dumps(self.admin_user))
+        # check status code
+        self.assertEqual(201, response.status_code)
+        expected = {
+            'message': 'Adminstrator Registration succesful, proceed to log in'
+        }
+        # check returned message
+        self.assertEqual(expected, json.loads(response.data))
+
+    def test_cant_register_same_user_twice(self):
+        response = self.client.post(
+            '/vi/auth/signup', data=json.dumps(self.new_user))
+        # check status code
+        self.assertEqual(201, response.status_code)
+        response = self.client.post(
+            '/vi/auth/signup', data=json.dumps(self.new_user))
+        # check status code
+        self.assertEqual(201, response.status_code)
+        response = self.client.post(
+            '/vi/auth/signup', data=json.dumps(self.new_user))
+        self.assertEqual(202, response.status_code)
+        expected = {'message': 'User already exists'}
+        self.assertEqual(expected, json.loads(response.data))
+
+    def test_user_login(self):
+        '''test user can login'''
+        # register user
+        response = self.client.post('/v1/auth/signup', data=json.dumps(
+            self.new_user))
+        # check user created succesfully
+        self.assertEqual(201, response.status_code)
+        # try login with right password
+        correct_password = self.new_user['password']
+        data = {'password': correct_password, 'email': self.new_user['email']}
+        response = self.client.post('/v1/auth/signin', data=json.dumps(data))
+        self.assertEqual(200, response.status_code)
+        expected = {'message': 'Login succesful'}
+        self.assertEqual(expected, json.loads(response.data))
+
+    def test_login_with_wrong_password(self):
+        '''test if user can log in with wrong password'''
+        # register user
+        response = self.client.post('/v1/auth/signup', data=json.dumps(
+            self.new_user))
+        # check user created succesfully
+        self.assertEqual(201, response.status_code)
+        wrong_password = 'wrongpassword'
+        data = {'password': wrong_password, 'email': self.new_user['email']}
+        response = self.client.post('/v1/auth/signin', data=json.dumps(data))
+        self.assertEqual(401, response.status_code)
+        expected = {
+            '''message': 'The email or password provided are incorrect.
+            Please try again'''}
+        self.assertEqual(expected, json.loads(response.data))
+
+
+class TestMealsManagement(BaseTestClass):
+    '''Test for MEal Resource'''
+    def test_get_all_meals(self):
```

Comment:
Add a test for getting only one meal(with the meal id)

Reply:
Done!